### PR TITLE
fix-addition-of-location-in-google-form

### PR DESCRIPTION
**Pull request by vldmrklnk**

### CODE REVIEW

File: InternshipBE/BL/Mapping/Profiles/CandidateProfile.cs [hunk 1]
```diff
@@ -88,12 +89,9 @@ public CandidateProfile()
 				.ForMember(entity => entity.EnglishLevelType, src => src.MapFrom(dto => Enum.Parse<EnglishLevelType>(dto.EnglishLevelType)))
 				.ForMember(entity => entity.IsPlanningToJoin, src => src.MapFrom(dto => dto.IsPlanningToJoin))
 				.ForMember(entity => entity.RegistrationDate, src => src.MapFrom(dto => dto.RegistrationDate))
-				.ForMember(entity => entity.Internship, o => o.Ignore())
-				.ForMember(entity => entity.Team, o => o.Ignore())
-				.ForMember(entity => entity.Users, o => o.Ignore())
-				.ForMember(entity => entity.Users, o => o.Ignore())
 				.ForMember(entity => entity.InternshipId, src => src.MapFrom(dto => dto.InternshipId))
-				.ForMember(entity => entity.TeamId, src => src.MapFrom(dto => dto.TeamId));
+				//.ForMember(entity => entity.TeamId, src => src.MapFrom(dto => dto.TeamId))
```
Comment (romanstat) [inline InternshipBE/BL/Mapping/Profiles/CandidateProfile.cs]: remove comment

File: InternshipBE/BL/Services/GoogleSheetService.cs [hunk 2]
```diff
@@ -6,28 +6,32 @@
...
 		private readonly IGoogleConfig _googleConfig;
 		private readonly ICandidateRepository _candidateRepository;
+		private readonly ILocationRepository _locationRepository;
 		private IMapper _mapper;
 		private SheetsService _sheetService;
 
 		private const int COLUMN_NAMES_ROW = 1;
 
-		public GoogleSheetService(IGoogleConfig googleConfig, IMapper mapper, ICandidateRepository candidateRepository)
+		public GoogleSheetService(IGoogleConfig googleConfig, IMapper mapper, ICandidateRepository candidateRepository, ILocationRepository locationRepository)
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleSheetService.cs]: unitofwork?

File: InternshipBE/BL/Services/GoogleSheetService.cs [hunk 3]
```diff
@@ -36,6 +40,8 @@ public async Task SaveNewCandidatesAsync()
 
 			if (values != null)
 			{
+				await SaveNewCountries(values);
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleSheetService.cs]: Async

File: InternshipBE/BL/Services/GoogleSheetService.cs [hunk 4]
```diff
@@ -73,6 +80,23 @@ private IList<IList<object>> GetValuesFromTable()
 
 			return response.Values;
 		}
+
+		private async Task SaveNewCountries(IList<IList<object>> values)
+		{
+			var names = await _locationRepository.GetAllNames().ToListAsync();
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleSheetService.cs]: ToListAsync in repo

File: InternshipBE/BL/Services/GoogleSheetService.cs [hunk 5]
```diff
@@ -73,6 +80,23 @@ private IList<IList<object>> GetValuesFromTable()
...
+			{
+				if (!names.Contains(value[LOCATION_OFFSET].ToString()))
+				{
+					newNames.Add(value[LOCATION_OFFSET].ToString());
+				}
+			}
+
+			var countries = _mapper.Map<List<Country>>(newNames.ToList());
+
+			await _locationRepository.RangeSaveAsync(countries);
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleSheetService.cs]: BulkSaveAsync

File: InternshipBE/BL/Services/GoogleSheetService.cs [hunk 6]
```diff
@@ -73,6 +80,23 @@ private IList<IList<object>> GetValuesFromTable()
...
+			var newNames = new HashSet<string>();
+			foreach (var value in values)
+			{
+				if (!names.Contains(value[LOCATION_OFFSET].ToString()))
+				{
+					newNames.Add(value[LOCATION_OFFSET].ToString());
+				}
+			}
+
+			var countries = _mapper.Map<List<Country>>(newNames.ToList());
```
Comment (romanstat) [inline InternshipBE/BL/Services/GoogleSheetService.cs]: try to refactor

File: InternshipBE/DAL/Interfaces/ILocationRepository.cs [hunk 7]
```diff
@@ -9,5 +10,7 @@ public interface ILocationRepository : IGenericRepository<Country>
         Task<List<Country>> GetLocationsAsync();
 
         Task<Country> CreateLocationAsync(string NameLocation);
+
+        IQueryable<string> GetAllNames();
```
Comment (romanstat) [inline InternshipBE/DAL/Interfaces/ILocationRepository.cs]: reimplement

**Review by romanstat — APPROVED** [review]

**Review by ihhyy — APPROVED** [review]

**Review by Xodoc — APPROVED** [review]

**Pull request merged**